internal/gh/pullrequest.go

```diff
@@ -211,3 +211,79 @@ func (c *Client) RepoPullRequests(ctx context.Context, opts RepoPullRequestOpts)
 		PullRequests: query.Repository.PullRequests.Nodes,
 	}, nil
 }
+
+type PullRequestByNumberInput struct {
+	Owner  string
+	Repo   string
+	Number int64
+}
+
+type Commit struct {
+	Oid string
+}
+
+func (c *Client) PullRequestMergeCommit(ctx context.Context, opts PullRequestByNumberInput) (*string, error) {
```

Comment:
It probably makes sense to just use the existing `Client.PullRequest` method for this and just unconditionally include this information.